### PR TITLE
docs : how to act on failure

### PR DESCRIPTION
**Pull request by julesbertrand**

### CODE REVIEW

File: docs/howto_act_on_failure.md [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+    with kfp.dsl.ExitHandler(notification_email_task, name="Exit Handler"):
+        # Add your pipeline tasks here.
+```
+
+### Slack notification using custom component
+
+You can also use a custom component to send notifications.
+Here is an example of a custom component that sends a notification to a Slack channel when a pipeline job ends.
+
+!!! success "Prerequisites"
```
Comment (hugovasselin) [inline docs/howto_act_on_failure.md:55]: this doesn't render properly for me: 

image

File: docs/howto_act_on_failure.md [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+    To use this component, you need to:
+
+    * Create a Slack app and a Slack webhook URL. [Official documentation](https://api.slack.com/messaging/webhooks).
+    * Create a secret and secret version in Secret Manager containing the Slack webhook URL. [Official documentation](https://cloud.google.com/secret-manager/docs/creating-and-accessing-secrets).
+
+
+
+=== "1. Create message to be sent to Slack"
+
+    Message cration includes:
```
Comment (hugovasselin) [inline docs/howto_act_on_failure.md:65]:
```suggestion
    Message creation includes:
```

**Review by hugovasselin — CHANGES_REQUESTED** [review]
Thanks @julesbertrand ! 

Maybe adding the `vertex_pipelines_notification_slack` can be added in a file in `assets` to lighthen a little bit the amount of Python code and make this function easier to re-use?

File: docs/howto_act_on_failure.md (hunk 1 above)
Comment (julesbertrand) [reply docs/howto_act_on_failure.md:55]: Do you use tech docs plugin to render it ? It's not pure markdown

**Comment by julesbertrand** [comment]
> Thanks @julesbertrand !
> 
> Maybe adding the `vertex_pipelines_notification_slack` can be added in a file in `assets` to lighthen a little bit the amount of Python code and make this function easier to re-use?

Found a middle ground: the code is in a python file, but I still display it from file using an extension that is part of techdocs-core. Might be useful for your bento PR @hugovasselin

**Review by hugovasselin — APPROVED** [review]

**Pull request merged**